### PR TITLE
Add slippi (a fork of Dolphin) v2.2.0

### PR DESCRIPTION
**Pull request by lukebrb**
Project slippi is a fork of Dolphin that implements rollback netplay.

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [✅ ] `brew cask audit --download {{cask_file}}` is error-free.
- [✅ ] `brew cask style --fix {{cask_file}}` reports no offenses.
- [✅ ] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [✅] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [✅ ] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [ ✅] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [ ✅] `brew cask install {{cask_file}}` worked successfully.
- [✅ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ✅] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ✅] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/slippi.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'slippi' do
+  version '2.2.0'
+  sha256 :no_check
```
Comment (miccal) [inline Casks/slippi.rb:3]: [The `sha256` needs to be set.](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md)

File: Casks/slippi.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (miccal) [inline Casks/slippi.rb:4]: Please add the following lone above the `url` stanza:
```
# github.com/project-slippi/Ishiiruka/ was verified as official when first introduced to the cask
```

File: Casks/slippi.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  url "https://github.com/project-slippi/Ishiiruka/releases/download/v#{version}/FM-Slippi-#{version}-Mac.zip"
+  appcast 'https://github.com/project-slippi/Ishiiruka/releases.atom'
+  name 'Project Slippi'
+  homepage 'https://slippi.gg'
```
Comment (miccal) [inline Casks/slippi.rb:8]: Please add a trailing slash `/`:
```
homepage 'https://slippi.gg/'
```

**Comment by miccal** [comment]
See https://slippi.gg/netplay for some justification of using `target:`.

**Comment by lukebrb** [comment]
Hi, the revisions are done — thank you for your thorough feedback!

> See https://slippi.gg/netplay for some justification of using `target:`.

As an addendum to this, since Project Slippi is a fork of Dolphin, users might already have the main Dolphin distribution installed on their machine. Using `target` allows us to get around this potential edge case, as users might want to have both editions of Dolphin installed.

File: Casks/slippi.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+cask 'slippi' do
```
Comment (vitorgalvao) [inline Casks/slippi.rb:1]: Does not follow the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md). Must be `project-slippi-dolphin`.

File: Casks/slippi.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  version '2.2.0'
+  sha256 'dcb9ce958b53a283894ef96b9ffeb159b680ca50c1adbddf2342884c7cd5ffe8'
+
+  # github.com/project-slippi/Ishiiruka/ was verified as official when first introduced to the cask
+  url "https://github.com/project-slippi/Ishiiruka/releases/download/v#{version}/FM-Slippi-#{version}-Mac.zip"
+  appcast 'https://github.com/project-slippi/Ishiiruka/releases.atom'
+  name 'Project Slippi'
```
Comment (vitorgalvao) [inline Casks/slippi.rb:8]: No “Project”.

File: Casks/slippi.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  homepage 'https://slippi.gg/'
+
+  app 'Dolphin.app', target: 'Slippi-Dolphin.app'
```
Comment (vitorgalvao) [inline Casks/slippi.rb:11]: Remove the `target`. This is a fork of a fork of a fork and doesn’t have many stars; it doesn’t warrant an exception to the token rules.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by lukebrb** [comment]
Done, thanks again!

**Comment by ran-dall** [comment]
@lukebrb Have you checked if this requires a `zap` stanza?

**Comment by lukebrb** [comment]
> @lukebrb Have you checked if this requires a `zap` stanza?

Good catch, didn't realize that Dolphin emu left cache artifacts. Added.

**Comment by lukebrb** [comment]
> @lukebrb Have you checked if this requires a `zap` stanza?

Good catch, added one based on the standard Dolphin emu cask. Thanks for your patience, everyone!

**Pull request merged**